sockets/lib/ipc.go

```diff
@@ -0,0 +1,125 @@
+/**
+ * IPC - Inter-Process Communication
+ * https://pokemonshowdown.com/
+ *
+ * This handles all communication between us and the parent process. The parent
+ * process creates a local TCP server using a random port. The port is passed
+ * down to us through the $PS_IPC_PORT environment variable. A TCP connection
+ * to the parent's server is created, allowing us to send messages back and
+ * forth.
+ */
+
+package sockets
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+)
+
+// This must be a byte that stringifies to either a hexadecimal or Unicode
+// escape code. Otherwise, it would be possible for someone to send a message
+// with the delimiter and break up messages.
+const DELIM byte = '\u0003'
```

Comment:
`byte` is not Unicode character, you may want to use `'\x03'` instead to clarify.